ci-operator/step-registry/cucushift/hypershift-extended/enable-dns/custom-kubeconfig/cucushift-hypershift-extended-enable-dns-custom-kubeconfig-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ ! -f "${SHARED_DIR}/kubeconfig" ]; then
+    exit 1
+fi
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+CLUSTER_NAME=$(oc get hostedclusters -n "$HYPERSHIFT_NAMESPACE" -o jsonpath='{.items[0].metadata.name}')
+echo "hostedclusters => ns: $HYPERSHIFT_NAMESPACE , cluster_name: $CLUSTER_NAME"
+if ! KAS_DNS_NAME=$(oc get "hostedclusters/${CLUSTER_NAME}" -n "${HYPERSHIFT_NAMESPACE}" \
+    -o jsonpath='{.spec.kubeAPIServerDNSName}' 2>/dev/null)
+then
+    echo "ERROR: HostedCluster '${CLUSTER_NAME}' not found in namespace '${HYPERSHIFT_NAMESPACE}'" >&2
+    exit 1
+elif [[ -z "${KAS_DNS_NAME}" ]]; then
+    echo "ERROR: KubeAPI Server DNS name not configured for '${CLUSTER_NAME}'" >&2
+    exit 1
+fi
+
+#Check secret with custom-kubeconfig generated in HC anc HCP namespaces
+if ! oc get -n "${HYPERSHIFT_NAMESPACE}" secret "${CLUSTER_NAME}-custom-admin-kubeconfig" &>/dev/null; then
```

Comment:
In the hc environment, the default hc kubeconfig will always be nested_kubeconfig. You could get it directly from the shared_dir